lib/config.js

```diff
@@ -229,6 +230,16 @@ function getLocalConfig(thisConfig, directory) {
             continue;
         }
 
+        // Check for root flag
+        if (localConfig.root === true) {
+            rootPath = path.dirname(localConfigFile);
+        }
+
+        // If root flag is set, don't consider file if it is above root
+        if (rootPath && localConfigFile.indexOf(rootPath) === -1) {
```

Comment:
Still a bit weary of this check. If there some way to be a bit smarter using the `path` module?